mailbox/store/src/test/java/org/apache/james/mailbox/store/search/AbstractMessageSearchIndexTest.java

```diff
@@ -1452,6 +1454,20 @@ public void searchShouldRetrieveExactlyMatchingMimeMessageID() throws Exception
             .containsOnly(m3.getUid());
     }
 
+    @Test
+    public void copiedMessageShouldAllBeIndexed() throws Exception {
+        MailboxPath newBoxPath = MailboxPath.forUser(USERNAME, "newBox");
+        MailboxId newBoxId = storeMailboxManager.createMailbox(newBoxPath, session).get();
+
+        storeMailboxManager.copyMessages(MessageRange.all(), inboxMessageManager.getId(), newBoxId, session);
+
+        SearchQuery searchQuery = new SearchQuery();
+
+        StoreMessageManager newBox = (StoreMessageManager) storeMailboxManager.getMailbox(newBoxId, session);
+        assertThat(messageSearchIndex.search(session, newBox.getMailboxEntity(), searchQuery))
```

Comment:
how do you ensure indexing is done when you run that query?

Reply:
I think I fixed your concern while trying to fix this test